experimental/traffic-portal/src/app/api/server.service.ts

```diff
@@ -186,4 +187,33 @@ export class ServerService extends APIService {
 
 		return this.put(`servers/${id}/status`, {offlineReason, status}).toPromise();
 	}
+
+	/**
+	 * Creating new Status.
+	 *
+	 * @param payload containes name and description for the status.
+	 * @returns The 'response' property of the TO status response. See TO API docs.
+	 */
+	public async createStatus(payload: RequestStatus): Promise<ResponseStatus> {
+		return this.post<ResponseStatus>("statuses", payload).toPromise();
+	}
+
+	/**
+	 * Updates status Details.
+	 *
+	 * @param payload containes name and description for the status., unique identifier thereof.
+	 * @param id The Status ID
+	 */
+	public async updateStatusDetail(payload: ResponseStatus, id: number): Promise<ResponseStatus> {
```

Comment:
this follows the pattern `{{operation}}{{thing}}Detail` but every other method is just `{{operation}}{{thing}}` - why?

Also note that a `ResponseStatus` includes an `id`, so you don't need the second argument here.

Reply:
changed